.github/workflows/build.yaml

```diff
@@ -201,6 +206,188 @@ jobs:
           name: ${{matrix.platform}}-lr-logs
           path: '*-logs.tar.gz'
 
+  libtester-deb-install-test:
+    name: libtester deb install test
+    needs: [d, Build, dev-package]
+    if: always() && needs.dev-package.result == 'success'
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu20, ubuntu22]
+    runs-on:  ["self-hosted", "enf-x86-lowtier"] # not sure if this warrants a better machine, but start here
+    container: ${{fromJSON(needs.d.outputs.p)[matrix.platform].image}}
+    steps:
+      - name: Download cdt
+        uses: AntelopeIO/asset-artifact-download-action@v2
+        with:
+          owner: AntelopeIO
+          repo: cdt
+          file: 'cdt_.*amd64.deb'
+          target: main
+          artifact-name: cdt_ubuntu_package_amd64
+          token: ${{github.token}}
+      - name: Download leap-dev
+        uses: actions/download-artifact@v3
+        with:
+          name: leap-dev-${{matrix.platform}}-amd64
+      - name: Install cdt and leap-dev Packages
+        run: |
+          apt install -y ./*.deb
+          rm ./*.deb
+      - name: checkout reference-contracts
+        uses: actions/checkout@v3
+        with:
+          repository: AntelopeIO/reference-contracts
+          path: reference-contracts
+      - name: Build & Test reference-contracts
+        run: |
+          cmake -S reference-contracts -B build -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=On
+          cmake --build build -- -j $(nproc)
+          cd build/tests
+          ctest --output-on-failure -j $(nproc)
+
+  libtester-make-dev-install-test:
+    name: libtester make dev-install test
+    needs: [d]
+    if: always() && needs.d.result == 'success'
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu20, ubuntu22]
+    runs-on: ["self-hosted", "enf-x86-beefy"]
+    container: ${{fromJSON(needs.d.outputs.p)[matrix.platform].image}}
+    steps:
+      - name: checkout leap
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          path: leap
+      - name: checkout cdt
+        uses: actions/checkout@v3
+        with:
+          repository: AntelopeIO/cdt
+          submodules: recursive
+          path: cdt
+      - name: checkout reference-contracts
+        uses: actions/checkout@v3
+        with:
+          repository: AntelopeIO/reference-contracts
+          path: reference-contracts
+      - name: Check directory structure
+        run: |
+          pwd
+          ls -l
+      - name: leap build and make dev-install
```

Comment:
we should try to figure out a way to do this without entirely building leap a second & third time.

For the build-dir case it should be easy -- we just package up the builddir from one job and extract it on the next job. We already package up the builddir as `ubuntu20-build` and `ubuntu22-build`.

The `dev-install` case is going to be trickier because the files will be co-mingled in `/usr`, and we don't really want to change `CMAKE_INSTALL_PREFIX` across the board. One option would be to perform the `dev-install` and then use `install_manifest_dev.txt` file as input to `tar -T`. Another option may be to enable `CPACK_ARCHIVE_COMPONENT_INSTALL` so a `leap-dev.tar.gz` is created.

Reply:
Yeah I was planning to come back to optimize once I had all three working. Thanks for the suggestions on where/how to start that process.  I'll definitely be making another pass once I get the 3 scenarios passing.

Reply:
Was able to get this to work with a clone of the `leap` repo along with the unpacked `builddir` then using the `install` and `dev-install` targets.